alchemist/alchemist-cognitive-agents/src/main/kotlin/it/unibo/alchemist/model/implementations/actions/utils/RandomGeneratorUtils.kt

```diff
@@ -0,0 +1,18 @@
+package it.unibo.alchemist.model.implementations.actions.utils
+
+import org.apache.commons.math3.random.RandomGenerator
+import java.util.*
+
+/**
+ * Generate a random double between the given bounds.
+ */
+fun RandomGenerator.nextDouble(from: Double, to: Double) = nextDouble() * (to - from) - to
+
+/**
+ * Fisher–Yates shuffle algorithm using Apache random number generator.
```

Comment:
Can you add a reference (ideally a doi or an rfc) for the algorithm?

Reply:
I can't find an "official" document reporting it, a deepening article like this one is good as well?
https://www.geeksforgeeks.org/shuffle-a-given-array-using-fisher-yates-shuffle-algorithm/

Reply:
It's in [a book from 1938](https://www.worldcat.org/title/statistical-tables-for-biological-agricultural-and-medical-research/oclc/14222135)
And it got enough dignity to have its own [Wikipedia page](https://en.wikipedia.org/wiki/Fisher–Yates_shuffle), report both, and we're done.